contracts/src/matching/HyperdriveMatchingEngine.sol

```diff
@@ -0,0 +1,472 @@
+        (, uint256 shortPaid) = _hyperdrive.openShort(
+            _order.amount,
+            _order.slippageGuard,
+            _order.minVaultSharePrice,
+            _order.options
+        );
+        if (_order.slippageGuard > shortPaid) {
+            _baseToken.safeTransfer(
+                _order.trader,
+                _order.slippageGuard - shortPaid
+            );
+        }
+    }
+
+    /// @dev Validates orders and returns their hashes.
+    /// @param _longOrder The order intent to open a long.
+    /// @param _shortOrder The order intent to open a short.
+    /// @param _addLiquidityOptions The options used when adding liquidity.
+    /// @param _removeLiquidityOptions The options used when removing liquidity.
+    /// @return longOrderHash The hash of the long order.
+    /// @return shortOrderHash The hash of the short order.
+    function _validateOrders(
+        OrderIntent calldata _longOrder,
+        OrderIntent calldata _shortOrder,
+        IHyperdrive.Options calldata _addLiquidityOptions,
+        IHyperdrive.Options calldata _removeLiquidityOptions
+    ) internal view returns (bytes32 longOrderHash, bytes32 shortOrderHash) {
+        // Ensure that the long and short orders are the correct type.
+        if (
+            _longOrder.orderType != OrderType.OpenLong ||
+            _shortOrder.orderType != OrderType.OpenShort
+        ) {
+            revert InvalidOrderType();
+        }
+
+        // Ensure that neither order has expired.
+        if (
+            _longOrder.expiry <= block.timestamp ||
+            _shortOrder.expiry <= block.timestamp
+        ) {
+            revert AlreadyExpired();
+        }
+
+        // Ensure that both orders refer to the same Hyperdrive pool.
+        if (_longOrder.hyperdrive != _shortOrder.hyperdrive) {
+            revert MismatchedHyperdrive();
+        }
+
+        // Ensure that all of the transactions should be settled with base.
+        if (
+            !_longOrder.options.asBase ||
+            !_shortOrder.options.asBase ||
+            !_addLiquidityOptions.asBase ||
+            !_removeLiquidityOptions.asBase
+        ) {
+            revert InvalidSettlementAsset();
+        }
+
+        // Ensure that the order's cross. We can calculate a worst-case price
+        // for the long and short using the `amount` and `slippageGuard` fields.
+        // In order for the orders to cross, the price of the long should be
+        // equal to or higher than the price of the short. This implies that the
+        // long is willing to buy bonds at a price equal or higher than the
+        // short is selling bonds, which ensures that the trade is valid.
+        if (
+            _longOrder.slippageGuard != 0 &&
+            _shortOrder.slippageGuard < _shortOrder.amount &&
+            _longOrder.amount.divDown(_longOrder.slippageGuard) <=
+            (_shortOrder.amount - _shortOrder.slippageGuard).divDown(
+                _shortOrder.amount
+            )
```

Comment:
Should this inequality just be `<` instead of `<=`? The way i'm reading it, if both sides are equal, then the long's max buy price equals the shorts min sell price, which should be a match no?

Reply:
Yeah, that's a good point